src/common/lustre_tools.c

```diff
@@ -1144,3 +1145,40 @@ ssize_t BuildLovEA(const entry_id_t *p_id, const attr_set_t *p_attrs,
     }
 }
 #endif
+
+/* from lustre/utils/lfs_project.c */
+static int project_get_xattr(const char *pathname, struct fsxattr *fsx)
+{
+        int ret, fd;
+
+        fd = open(pathname, O_RDONLY | O_NOCTTY | O_NDELAY);
```

Comment:
To avoid any useless lock or data prefetching, I suggest adding the O_PATH flag here. Could you try and see if this still works?

Reply:
thanks will do and report back

Reply:
Finally, I tested this!!

If I add `O_PATH` here, the following ioctl:
```
ret = ioctl(fd, LL_IOC_FSGETXATTR, fsx);
```
fails with EBADF

```
2022/07/08 09:57:16 robinhood@oak-rbh02[154119/2] FS_Scan | Starting scan of /oak
2022/07/08 09:57:16 robinhood@oak-rbh02[154119/3] projid | Error: project_get_xattr: failed to get xattr for '/oak/.nonreg': Bad file descriptor
2022/07/08 09:57:16 robinhood@oak-rbh02[154119/3] FS_Scan | Failed to get lustre projid for dir /oak/.nonreg: error -9
```

Reply:
It looks like O_NDELAY == O_NONBLOCK so it's already there. I've been scanning our filesystem with the latest version of this patch and the scan goes up to 2,800 entries/sec, which I think is pretty good for HDD based OSTs.

Reply:
BTW, I used the same open flags than in `lustre/utils/lfs_project.c`, that is:
```
114         fd = open(pathname, O_RDONLY | O_NOCTTY | O_NDELAY);
```